driver/glu_driver.hpp

```diff
@@ -0,0 +1,486 @@
+    miopenTensorDescriptor_t outputTensor;
+
+    // Backwards
+    miopenTensorDescriptor_t inputTensorGrad;
+    miopenTensorDescriptor_t outputTensorGrad;
+
+    std::unique_ptr<GPUMem> in_dev;
+    std::unique_ptr<GPUMem> out_dev;
+
+    std::unique_ptr<GPUMem> inGrad_dev;
+    std::unique_ptr<GPUMem> outGrad_dev;
+
+    std::vector<Tgpu> in;
+    std::vector<Tgpu> out;
+    std::vector<Tref> outhost;
+
+    std::vector<Tgpu> inGrad;
+    std::vector<Tgpu> outGrad;
+    std::vector<Tref> inGradhost;
+
+    uint32_t dim;
+};
+
+template <typename Tgpu, typename Tref>
+int GLUDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int GLUDriver<Tgpu, Tref>::GetandSetData()
+{
+    SetBNParametersFromCmdLineArgs();
+
+    std::vector<int> in_len = inflags.GetValueTensor("dim-lengths").lengths;
+    dim                     = inflags.GetValueInt("DimToSplit");
+
+    SetTensorNd(inputTensor, in_len, data_type);
+
+    std::vector<int> out_len;
+
+    for(int i = 0; i < in_len.size(); i++)
+    {
+        if(i != dim)
+        {
+            out_len.push_back(in_len[i]);
+        }
+        else
+        {
+            out_len.push_back(in_len[i] / 2);
+        }
+    }
+
+    if(out_len.empty())
+        out_len.push_back(1);
+
+    SetTensorNd(outputTensor, out_len, data_type);
+
+    // Backwards
+    SetTensorNd(inputTensorGrad, in_len, data_type);
+    SetTensorNd(outputTensorGrad, out_len, data_type);
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int GLUDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw",
+                         'F',
+                         "1",
+                         "Run only Forward (1) or Run both Forward and Backward (0) (Default=1)",
+                         "int");
+    inflags.AddTensorFlag(
+        "dim-lengths", 'D', "256x512", "The dimensional lengths of the input tensor");
+    inflags.AddInputFlag(
+        "DimToSplit", 'R', "0", "The indice of the dimensions to be split half(Default=0)", "int");
+    inflags.AddInputFlag("iter", 'i', "10", "Number of Iterations (Default=10)", "int");
+    inflags.AddInputFlag("verify", 'V', "1", "Verify Each Layer (Default=1)", "int");
+    inflags.AddInputFlag("time", 't', "0", "Time Each Layer (Default=0)", "int");
+    inflags.AddInputFlag(
+        "wall", 'w', "0", "Wall-clock Time Each Layer, Requires time == 1 (Default=0)", "int");
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int GLUDriver<Tgpu, Tref>::SetBNParametersFromCmdLineArgs()
+{
+    forw = inflags.GetValueInt("forw");
+    if(forw != 0 && forw != 1)
+    {
+        printf("Incorrect Forward Mode\n");
+        exit(EXIT_FAILURE); // NOLINT (concurrency-mt-unsafe)
```

Comment:
Let's remove exit() and use proper return code propagation/check

Reply:
updated it.